<?xml version="1.0"?>
<robot name="vtec_u3" xmlns:xacro="http://www.ros.org/wiki/xacro" >
  <xacro:arg name="debug" default="0"/>
  <xacro:arg name="namespace" default="vtec_u3"/>
  <xacro:arg name="inertial_reference_frame" default="world_ned"/>

  <xacro:include filename="$(find vehicle_descriptions)/urdf/UUVs/vtec_u3_base.xacro"/>
  <xacro:include filename="$(find vehicle_descriptions)/urdf/UUVs/vtec_u3_sensors.xacro"/> 
  <xacro:include filename="$(find vehicle_descriptions)/urdf/UUVs/vtec_u3_arm.xacro"/> 
  <xacro:include filename="$(find vehicle_descriptions)/urdf/UUVs/vtec_u3_shooter.xacro"/> 

  <virtual_joint name="world_joint" type="planar" parent_frame="world_ned" child_link="vtec_u3_base_link"/>

  <xacro:vtec_u3_base namespace="$(arg namespace)"/>
  <xacro:vtec_u3_sensors namespace="$(arg namespace)"/>
  <xacro:vtec_u3_arm parent="$(arg namespace)"/>
  <xacro:vtec_u3_shooter parent="$(arg namespace)"/>
  <!-- Joint state publisher plugin -->
  <!-- <xacro:default_joint_state_publisher namespace="$(arg namespace)" update_rate="10"/> -->
</robot>